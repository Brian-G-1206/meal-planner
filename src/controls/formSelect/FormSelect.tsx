import { Field } from "formik"
import React, { FunctionComponent } from "react"

export interface FormSelectProps {
	fieldName: string
	options: FormSelectOption[]
}

export interface FormSelectOption {
	label: string
	value: string
}

export const FormSelect: FunctionComponent<FormSelectProps> = ( props: FormSelectProps ) => 
<Field as="select" name={props.fieldName} >
	{props.options.map(( opt, key ) => <option key={key} value={opt.value}>{opt.label}</option>)}
</Field>