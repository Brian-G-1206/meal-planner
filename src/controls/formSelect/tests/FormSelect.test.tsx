import { Formik } from 'formik'
import renderer from 'react-test-renderer'
import { FormSelect } from '../FormSelect'

describe('src/controls/formSelect', () => {
	it('should match snapshot', () => {
		// TODO: this cannot be the right way to do this, but it's "working" ...
		let Subject = renderer
			.create(
				<Formik initialValues={{fieldName: ''}} onSubmit={() => {}} >
					<FormSelect fieldName="fieldName" options={[ {value: "1", label: "One"}, {value: "1", label: "One"} ]} />
				</Formik>)
			.toJSON()
		expect(Subject).toMatchSnapshot()
	})
})
