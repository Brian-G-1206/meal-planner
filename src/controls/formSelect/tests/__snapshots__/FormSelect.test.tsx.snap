// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/controls/formSelect should match snapshot 1`] = `
<select
  name="fieldName"
  onBlur={[Function]}
  onChange={[Function]}
  value=""
>
  <option
    value="1"
  >
    One
  </option>
  <option
    value="1"
  >
    One
  </option>
</select>
`;
